types/epilogue/index.d.ts

```diff
@@ -0,0 +1,187 @@
+    app: Express;
+    sequelize: Sequelize;
+    base?: string;
+    updateMethod?: string;
+}
+
+export interface BaseContollerOptions {
+    endpoint: string;
+    model: any;
+    app: Express;
+    resource: Resource;
+    include: Array<{ model: any } | string>;
+}
+
+export interface Context {
+    instance: Resource;
+    criteria: any;
+    attributes: any;
+    options: any;
+    continue: () => void;
+    skip: () => void;
+    stop: () => void;
+    error: (status: number | EpilogueError, message?: string, errorList?: string[], cause?: Error) => void;
+}
+
+export class BaseController {
+    constructor(options: BaseContollerOptions);
+    endpoint: Endpoint;
+    model: any;
+}
+
+export class CreateController extends BaseController {
+    write: (req: Request, res: Response, context: Context) => Promise<() => void>;
+}
+
+export class ReadController extends BaseController {
+    fetch: (req: Request, res: Response, context: Context) => Promise<() => void>;
+}
+
+export class UpdateController extends BaseController {
+    fetch: (req: Request, res: Response, context: Context) => Promise<() => void>;
+    write: (req: Request, res: Response, context: Context) => Promise<() => void>;
+}
+
+export class DeleteController extends BaseController {
+    fetch: (req: Request, res: Response, context: Context) => Promise<() => void>;
+    write: (req: Request, res: Response, context: Context) => Promise<() => void>;
+}
+
+export class ListController extends BaseController {
+    fetch: (req: Request, res: Response, context: Context) => Promise<() => void>;
+    _safeishParse: (value: any, type: DataTypeAbstract | DataTypeString | DataTypeChar | DataTypeText | DataTypeNumber |
+                        DataTypeInteger | DataTypeBigInt | DataTypeFloat | DataTypeTime | DataTypeDate | DataTypeDateOnly |
+                        DataTypeBoolean | DataTypeNow | DataTypeBlob | DataTypeDecimal | DataTypeUUID | DataTypeUUIDv1 |
+                        DataTypeUUIDv4 | DataTypeHStore | DataTypeJSONType | DataTypeJSONB | DataTypeVirtual |
+                        DataTypeArray | DataTypeEnum | DataTypeRange | DataTypeReal | DataTypeDouble | DataTypeGeometry,
+                    sequelize: Sequelize) => any;
+}
+
+export interface ResourceOptions {
+    model: any;
+    endpoints: string[];
+    actions?: string[];
+    include?: Array<{ model: any } | string>;
+    pagination?: boolean;
+    search?: ResourceSearchOption;
+    sort?: ResourceSortOption;
+    reloadInstances?: boolean;
+    associations?: AssociationOptions;
+    excludeAttributes?: string[];
+    readOnlyAttributes?: string[];
+    updateMethod?: string;
+}
+
+export class EpilogueError extends Error {
+    name: string;
+    message: string;
+    errors: string[];
+    status: EpilogueError | number;
+    cause: Error;
+}
+
+export function initialize(options: InitializeOptions): undefined;
+export function resource(options: ResourceOptions): Resource;
```

Comment:
The `options` seems to be optional.
https://github.com/dchester/epilogue/blob/master/lib/index.js#L14
https://github.com/dchester/epilogue/blob/master/lib/index.js#L40